### PR TITLE
Commited an axum version of the tutorial!

### PR DESCRIPTION
**Pull request by benwis**
Hiya!
I made a version of the tutorial that uses Axum instead of tide! I also reorganized the tutorial folder to allow different server implementations.

I'd recommend adding a section about the tutorial to the README, but I didn't do that here.

Fixes #
- Added axum version of tutorial site
- Structured tutorial folder a bit
- [x] cargo fmt has been run
- [x] cargo test has been run and passes
- [x] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

File: tutorial/server/axum/src/auth.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+use crate::startup::AppState;
+use axum::{
+    extract::{Extension, Json, Path},
+    http::StatusCode,
+    response::{IntoResponse, Response},
+};
+use axum_sessions::async_session::Session;
+use webauthn_rs::prelude::*;
+
+// 3. The first step a client (user) will carry out is requesting a credential to be
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply tutorial/server/axum/src/auth.rs:11]: Ahhh I see. You might want to add "for steps 1 to 2 see X file ..."

File: tutorial/server/axum/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+use rand::prelude::*;
+
+#[macro_use]
+extern crate tracing;
+
+mod auth;
+mod startup;
+
+// 7. That's it! The user has now authenticated!
```
Comment (Firstyear) [inline tutorial/server/axum/src/main.rs:32]: Seems a bit out of place?

File: tutorial/server/axum/src/startup.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        let rp_id = "localhost";
+        // Url containing the effective domain name
+        // MUST include the port number!
+        let rp_origin = Url::parse("http://localhost:8080").expect("Invalid URL");
+        let builder = WebauthnBuilder::new(rp_id, &rp_origin).expect("Invalid configuration");
+
+        // Now, with the builder you can define other options.
+        // Set a "nice" relying party name. Has no security properties and
+        // may be changed in the future.
+        let builder = builder.rp_name("Panda5");
```
Comment (Firstyear) [inline tutorial/server/axum/src/startup.rs:39]: Should the rp_name here be "Webauthn-RS Axum Demo"?

**Review by Firstyear — CHANGES_REQUESTED** [review]
looks really good! Not too much in the review, just some cleanup. Thank you so much! Please make sure you add yourself to the CONTRIBUTORS file as well :)

File: tutorial/server/axum/src/auth.rs [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+            .copied()
+            .unwrap_or_else(Uuid::new_v4)
+    };
+
+    // Remove any previous registrations that may have occured from the session.
+    session.remove("reg_state");
+
+    // If the user has any other credentials, we exclude these here so they can't be duplicate registered.
+    // It also hints to the browser that only new credentials should be "blinked" for interaction.
+    //TODO: Store registered User IDs and keys in the DB
```
[the remark this answers is not in the document]
Comment (benwis) [reply tutorial/server/axum/src/auth.rs:77]: I have some todos added where I'd add in a db and functions to handle it. This is one of mine

File: tutorial/server/axum/src/auth.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (benwis) [reply tutorial/server/axum/src/auth.rs:11]: Yeah, I kept all your comments but they stayed with the associated functions. I'll clarify it

File: tutorial/server/axum/src/startup.rs (hunk 3 above)
Comment (benwis) [reply tutorial/server/axum/src/startup.rs:39]: Yes it should

File: tutorial/server/axum/src/main.rs (hunk 2 above)
Comment (benwis) [reply tutorial/server/axum/src/main.rs:32]: That's what I thought, but the tide version has it in the same spot above index_view

**Comment by benwis** [comment]
I think I've addressed your concerns, let me know if I missed anything!

File: tutorial/server/axum/src/main.rs (hunk 2 above)
Comment (Firstyear) [reply tutorial/server/axum/src/main.rs:29]: Ahhhh I see. :)

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
Thank you so much @benwis!

**Pull request merged**